packages/wonder-blocks-form/src/components/checkbox.tsx

```diff
@@ -6,9 +6,9 @@ import ChoiceInternal from "./choice-internal";
 // Keep synced with ChoiceComponentProps in ../util/types.js
 type ChoiceComponentProps = AriaProps & {
     /**
-     * Whether this component is checked
+     * Whether this component is checked or indeterminate
      */
-    checked: boolean;
+    checked: boolean | "mixed";
```

Comment:
When `checked: boolean | null | undefined` in `checkbox.tsx`, I get type errors that say `checked` is incompatible: 

<img width="698" alt="Screenshot 2023-03-31 at 10 27 18 AM" src="https://user-images.githubusercontent.com/100858764/229148264-0c5e4ad7-7622-46aa-a107-940233ff1af3.png">
<img width="704" alt="Screenshot 2023-03-31 at 10 28 30 AM" src="https://user-images.githubusercontent.com/100858764/229148588-e46bb11a-221c-40bb-a4bc-b1ca02db1808.png">

I changed the `checked` type to match in `choice.tsx`, `choice-internal.tsx`, and `types.ts` and the error didn't go away.

Also recieved type error in `checkbox-core.tsx` for `StyledInput` `checked` prop that led to `index.d.ts` for `InputHTMLAttributes` where `checked?: boolean | "mixed" | undefined`.

<img width="684" alt="Screenshot 2023-03-31 at 10 29 59 AM" src="https://user-images.githubusercontent.com/100858764/229149001-ec3f7176-61ee-4d0f-a60c-ff2d762fea28.png">

In Storybook:

The indeterminate icon appears when `checked={null}` or `checked="mixed"`, but doesn't appear when `checked={undefined}`.

The `aria-checked` doesn't appear in the DOM when the `checked` value is `null` or `undefined`, only for `checked="mixed"`

I could totally by missing something or going about this wrong! This is just the scavenger hunt that led me to settle on `boolean | "mixed"`, but I agree that `null`/`undefined` feels like a better approach.

Reply:
ahh ok, thanks for all the context, it's super helpful :).

So the issue that you experienced was because, I suspect there's one missing step to be able to use `null/undefined` in checkboxes.

The `props.checked` value can still have `true | false | null | undefined`, but we'd need to map that value to `aria-checked` so that ARIA attribute can have the right value and avoid that TS error.

I could imagine something like this would work (not fully sure)...

Just adding a snippet for illustration purposes:

```
type Checked = boolean | null | undefined;
type AriaChecked = "true" | "false" | "mixed";

function mapCheckedToAriaChecked(value: Checked): AriaChecked {
  switch(value) {
    case true:
      return "true";
    case false:
      return "false";
    default:
      return "mixed";
  }
}

// then in the component
const ariaChecked = mapCheckedToAriaChecked(props.checked);
...
return (
  <input type="checkbox" aria-checked={ariaChecked} ... />
);
```

What do you think? happy to chat more about it.

Reply:
Oh awesome, thank you Juan! I will try this and check back to see about pairing 😊

Reply:
So that worked! The only issue is that `undefined` sets `aria-checked` to `false` instead of `"mixed"` even though it "looks" right.